lib/utils/getOverlayEntry.js

```diff
@@ -0,0 +1,57 @@
+const querystring = require('querystring');
+
+/**
+ * Creates the webpack entry for the overlay
+ * @param {import('../types').NormalizedPluginOptions} options Configuration options for this plugin.
+ * @param {import('webpack').Compiler.options.devServer} devServer The webpack devServer config
+ * @returns {WebpackEntry[]} An array of webpack entries; first is the prepend entry, and second is the overlay entry
+ */
+function getOverlayEntry(options, devServer) {
+  /** @type {Record<string, *>} */
+  let resourceQuery = {};
+
+  if (Object.keys(devServer).length) {
```

Comment:
Do we need this check since all the destructured parameters will be `undefined`?

Reply:
Or even we could just not pass a default empty object down, so we could check for nullish behaviour here?